### ISSUE TITLE
Choose strategy for looking up async hasMany relationships

### ISSUE THREAD

**Issue by g-cassie** [dustinfarris/ember-django-adapter#32]
TLDR: determine whether and how to support optimized queries for async hasMany relationships.

In Toran's original ED adapter, hasMany records were looked up using a nested url structure (e.g. `/dealerships/1/cars/`.  To my knowledge this was a design decision made by Toran that is not mandated (or suggested) by either ED or DRF.  

ED introduced the `coalesceFindIds` option in beta.9 as its solution for getting hasMany attributes.  With this option enabled, whenever the adapter needs to lookup at an async hasMany relationship, it will take the list of related ids and generate a request in the format `/cars?ids[]=1&ids[]=2&ids[]=3`.  This format of request is not supported by DRF by default ([though it can be implemented](https://github.com/g-cassie/ember-drf/blob/master/ember_drf/filters.py)).

ED continues to support a set of adapter methods that make it possible to direct async hasMany look ups to particular urls (see [here](http://emberjs.com/api/data/classes/DS.Adapter.html#method_groupRecordsForFindMany) and [here](http://emberjs.com/api/data/classes/DS.Adapter.html#method_findMany).  This makes it possible to cause an ED adapter to use a url structure like `/dealerships/1/cars/` (like Toran's adapter).  

Using DRF viewsets, an endpoint for the above url could be implement like this:

```python
class DealershipViewset(viewsets.ModelViewset):

    @detail_view(methods=['get'])
    def cars(self, request):
         queryset = self.get_object().cars.all()
         serializer = CarSerializer(queryset, many=True)
         return serializer.data
```

If this approach is taken, a decision would need to be made on how urls for multiword relationships should be formatted, noting that DRF currently does not support dasherized urls for `@detail_view` and `@list_view endpoints`. 

See https://github.com/dustinfarris/ember-django-adapter/issues/19 for some other lead up discussion.

So hopefully this frames the problem well. I should mention that my interest in this is purely academic as I have switched to using a DRF side adapter and pure ember data.  None the less, I am happy to stay involved in this discussion.
